tests/src/system/packages/MessagingFeedTests.scala

```diff
@@ -0,0 +1,190 @@
+    val sessionTimeout = 10 seconds
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk()
+
+    val messagingPackage = "/whisk.system/messaging"
+    val kafkaFeed = "kafkaFeed"
+    val messageHubFeed = "messageHubFeed"
+
+    behavior of "Kafka connector"
+
+    it should "fire a trigger when a message is posted to Kafka" in withAssetCleaner(wskprops) {
+        val openwhisk_home = System.getenv("OPENWHISK_HOME")
+        val file = new File(openwhisk_home + "/whisk.properties")
+        val config = new WhiskConfig(Map("kafka.host" -> null, "kafka.host.port" -> null),
+            Set(), file)
+        // Check if the Kafka host and port are valid in the property file.
+        assert(config.isValid)
+        implicit val ec = ExecutionContextFactory.makeCachedThreadPoolExecutionContext()
+        val producer = new KafkaProducerConnector(config.kafkaHost, ec)
+        (wp, assetHelper) =>
+            val triggerName = s"dummyKafkaTrigger-${System.currentTimeMillis}"
+
+            // Make sure the package of the Kafka feed exists.
+            val packageGetResult = wsk.pkg.get(messagingPackage)
+            packageGetResult.stdout should include("ok")
+
+            // Create a trigger with the Kafka feed.
+            var kafka_brokers = JsArray(JsString(config.kafkaHost))
+            val feedCreationResult = assetHelper.withCleaner(wsk.trigger, triggerName) {
+                (trigger, name) =>
+                    trigger.create(name, feed = Some(s"$messagingPackage/$kafkaFeed"), parameters = Map(
+                        "brokers" -> kafka_brokers,
+                        "topic" -> kafka_topic.toJson))
+            }
+            feedCreationResult.stdout should include("ok")
+            
+            // Give the consumer some to initialize.
+            Thread.sleep(2000)
+
+            // Send a message to the Kafka.
+            val message = new Message { override val serialize = s"${System.currentTimeMillis}" }
+            val sent = Await.result(producer.send(kafka_topic, message), 10 seconds)
+            
+            // Check if the trigger is fired.
+            val activations = wsk.activation.pollFor(N = 1, Some(triggerName), retries = 30).length
+            withClue("Change feed trigger count: ") { activations should be(1) }
+            producer.close()
+    }
+    
+    it should "fire a trigger when a message is posted to the message hub" in withAssetCleaner(wskprops) {
+        // Load the credentials from the vcap file.
+        var credentials = TestUtils.getCredentials("message_hub")
+        val user = credentials.get("user").getAsString()
+        val password = credentials.get("password").getAsString()
+        val kafka_admin_url = credentials.get("kafka_admin_url").getAsString()
+        val api_key = credentials.get("api_key").getAsString()
+        val kafka_brokers_sasl_json_array = credentials.get("kafka_brokers_sasl").getAsJsonArray()
+
+        var vec = Vector[JsString]()
```

Comment:
This array juggling is pretty messy, but I assume it is unavoidable. Let's move this out somewhere that it can be reused. Perhaps a MessageHubCredentials class that does all this work in a `getKafkaBrokersSasl()` method that returns whatever type is directly needed for use in `trigger.create()`.